discussions/Python Package Managers.md

```diff
@@ -0,0 +1,52 @@
+
+However, more complex projects may benefit from package managers that provide addtional automation for building, testing, and distributing the package. Until recently, [Poetry](https://python-poetry.org/) was a popular choice for this purpose, but the new [UV](https://docs.astral.sh/uv/intro/) package manager is gaining traction. Recent DSST members have reported bugs in Poetry projects in recent weeks.
+
+ These tools provide a lot of automation for managing dependencies, building, testing, and distributing the package. UV is written in Rust, which is a compiled language, so it is faster than pip and other Python package managers, provides lock-file support, and is compatible with existing Python projects such as the pyproject.toml file format. There is a [PYPI package](https://pypi.org/project/poetry-to-uv/) for converting Poetry projects to UV projects.
+
+If a project requires custom binaries for execution or has multiple languages, then a package manager like Conda is a good choice. Conda is a package manager for any language, but it is particularly popular in the life sciences. Conda can be used to create an environment with the necessary dependencies to run the package. Recent additions to the Conda ecosystem are [mamba/micromamba](https://mamba.readthedocs.io/en/latest/) (a drop-in replacement for Conda) and [Pixi](https://prefix.dev/) which currently uses UV. With each claiming to be faster than Conda.
+
+### End Users
+
+If end users are not Python developers, then ease of installation is important. More automated tools like Conda or UV are better suited for this purpose, as they facilitate the installation of dependencies. However, if the end users are not technical, then the project may be better suited being placed in a containerized environment like [Docker](https://www.docker.com/)/[Podman](https://podman.io/) or [Singularity](https://sylabs.io/singularity/). Finally, if the end users are interested in learning Python, starting out with simple tools like pip and venv is a good way to demonstrate why they should use a more complex tool like Conda or UV as they get more comfortable with Python.
+
+Note, while many end users are permitted to install Python, they may not be permitted to install Docker or Singularity. In this case, a package manager like Conda or UV is a good choice.
+
+### Setup complexity
+
+While a containerized application is easy to use, setting up the container can be complex depending on the system. For example, if the system does not have Docker or Singularity installed, then the user will need to install it. This can be a barrier to entry for some users lacking the expertise in containerization or Docker/Singularity may not allowed by the system administrator.
```

Comment:
```suggestion
While a containerized application is easy to use, setting up the container can be complex depending on the system. For example, if the system does not have Docker or Singularity installed, then the user will need to install it. This can be a barrier to entry for some users lacking the expertise in containerization. Or Docker/Singularity may not be allowed by the system administrator.
```